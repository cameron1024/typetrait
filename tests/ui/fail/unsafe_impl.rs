fn main() {}

typetrait::union! {
    Status = Enabled | Disabled
}

struct Illegal;
impl Status for Illegal {}
