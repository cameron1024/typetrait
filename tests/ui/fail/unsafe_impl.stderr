error[E0277]: the trait bound `Illegal: StatusSealed` is not satisfied
 --> tests/ui/fail/unsafe_impl.rs:8:6
  |
8 | impl Status for Illegal {}
  |      ^^^^^^ the trait `StatusSealed` is not implemented for `Illegal`
  |
note: required by a bound in `Status`
 --> tests/ui/fail/unsafe_impl.rs:4:5
  |
4 |     Status = Enabled | Disabled
  |     ^^^^^^ required by this bound in `Status`
